lib-ee/emqx_ee_connector/src/emqx_ee_connector_sqlserver.erl

```diff
@@ -0,0 +1,526 @@
+-type common_reason() :: connection_closed | extended_error() | term().
+%% extended error type with ODBC
+%% and native database error codes, as well as the base reason that would have been
+%% returned had extended_errors not been enabled.
+-type extended_error() :: {string(), integer(), _Reason :: term()}.
+%% Name of column in the result set
+-type col_name() :: string().
+%% e.g. a list of the names of the selected columns in the result set.
+-type col_names() :: [col_name()].
+%% A column value.
+-type value() :: null | term().
+%% A list of rows from the result set.
+-type rows() :: [row()].
+
+%% -type result_tuple() :: {updated, n_rows()} | {selected, col_names(), rows()}.
+-type updated_tuple() :: {updated, n_rows()}.
+-type selected_tuple() :: {selected, col_names(), rows()}.
+%% The number of affected rows for UPDATE,
+%% INSERT, or DELETE queries. For other query types the value
+%% is driver defined, and hence should be ignored.
+-type n_rows() :: integer().
+
+%% These type was not used in this module, but we may use it later
+%% -type odbc_data_type() ::
+%%     sql_integer
+%%     | sql_smallint
+%%     | sql_tinyint
+%%     | {sql_decimal, precision(), scale()}
+%%     | {sql_numeric, precision(), scale()}
+%%     | {sql_char, size()}
+%%     | {sql_wchar, size()}
+%%     | {sql_varchar, size()}
+%%     | {sql_wvarchar, size()}
+%%     | {sql_float, precision()}
+%%     | {sql_wlongvarchar, size()}
+%%     | {sql_float, precision()}
+%%     | sql_real
+%%     | sql_double
+%%     | sql_bit
+%%     | atom().
+%% -type precision() :: integer().
+%% -type scale() :: integer().
+%% -type size() :: integer().
+
+-type state() :: #{
+    poolname := binary(),
+    resource_opts := map(),
+    sql_templates := map()
+}.
+
+%%====================================================================
+%% Configuration and default values
+%%====================================================================
+
+roots() ->
+    [{config, #{type => hoconsc:ref(?MODULE, config)}}].
+
+fields(config) ->
+    [
+        {server, server()}
+        | add_default_username(emqx_connector_schema_lib:relational_db_fields())
+    ].
+
+add_default_username(Fields) ->
+    lists:map(
+        fun
+            ({username, OrigUsernameFn}) ->
+                {username, add_default_fn(OrigUsernameFn, <<"sa">>)};
+            (Field) ->
+                Field
+        end,
+        Fields
+    ).
+
+add_default_fn(OrigFn, Default) ->
+    fun
+        (default) -> Default;
+        (Field) -> OrigFn(Field)
+    end.
+
+server() ->
+    Meta = #{desc => ?DESC("server")},
+    emqx_schema:servers_sc(Meta, ?SQLSERVER_HOST_OPTIONS).
+
+%%====================================================================
+%% Callbacks defined in emqx_resource
+%%====================================================================
+
+callback_mode() -> async_if_possible.
+
+is_buffer_supported() -> false.
+
+on_start(
+    BridgeId = PoolName,
```

Comment:
Nit: this is actually a `manager_id()` (from `emqx_resource.hrl`), which is different from what we get from `emqx_bridge_resource:bridge_id/2`.  So the variable name might be a bit confusing.